supervisor/index.js

```diff
@@ -27,26 +27,30 @@ class Supervisor {
     }
   }
 
-  // TODO
   getProcessByThreadId(id) {
-    return id;
+    for (const name in this.processes) {
+      if (this.processes[name].threadId === id) {
+        return this.processes[name];
+      }
+    }
   }
 
   send(recipient, payload) {
+    let process = null;
+
     if (typeof recipient === 'number') {
-      for (const name in this.processes) {
-        if (this.processes[name].threadId === recipient) {
-          this.processes[name].postMessage(payload);
-          return true;
-        }
-      }
+      process = this.getProcessByThreadId(recipient);
     }
 
     if (typeof recipient === 'string' && this.isNamePresent(recipient)) {
-      const process = this.getProcess(recipient);
+      process = this.getProcess(recipient);
+    }
+
+    if (process) {
       process.postMessage(payload);
       return true;
     }
+
```

Comment:
I'd love to avoid `let`ting and reassigning, it's a pattern that I don't like and it reads to less declarative code. But kudos to you finding a way to refactor the garbage I wrote!

I have no idea how to rewrite this now, I hope I'll come up with something later :)